Index: serial-test/Cargo.toml
===================================================================
--- serial-test.orig/Cargo.toml
+++ serial-test/Cargo.toml
@@ -52,10 +52,6 @@ path = "src/lib.rs"
 name = "tests"
 path = "tests/tests.rs"
 
-[dependencies.document-features]
-version = "0.2"
-optional = true
-
 [dependencies.env_logger]
 version = ">=0.6.1"
 optional = true
@@ -107,7 +103,6 @@ default = [
     "logging",
     "async",
 ]
-docsrs = ["dep:document-features"]
 file_locks = ["dep:fslock"]
 logging = ["dep:log"]
 test_logging = [
