Index: serial-test/Cargo.toml
===================================================================
--- serial-test.orig/Cargo.toml
+++ serial-test/Cargo.toml
@@ -57,12 +57,6 @@ version = ">=0.6.1"
 optional = true
 default-features = false
 
-[dependencies.fslock]
-version = "0.2"
-features = ["std"]
-optional = true
-default-features = false
-
 [dependencies.futures]
 version = "^0.3"
 features = ["executor"]
@@ -103,7 +97,6 @@ default = [
     "logging",
     "async",
 ]
-file_locks = ["dep:fslock"]
 logging = ["dep:log"]
 test_logging = [
     "logging",
