Index: asn1-rs/Cargo.toml
===================================================================
--- asn1-rs.orig/Cargo.toml	2024-07-13 23:07:29.663763368 +0200
+++ asn1-rs/Cargo.toml	2024-07-13 23:08:52.060535221 +0200
@@ -114,7 +114,7 @@
 [features]
 bigint = ["num-bigint"]
 bits = ["bitvec"]
-datetime = ["time"]
+datetime = ["std","time"]
 default = ["std"]
 serialize = ["cookie-factory"]
 std = []
