Index: sequoia-openpgp/Cargo.toml
===================================================================
--- sequoia-openpgp.orig/Cargo.toml
+++ sequoia-openpgp/Cargo.toml
@@ -58,15 +58,15 @@ path = "tests/secret-leak-detector/detec
 name = "run_benchmarks"
 harness = false
 
-[dependencies.aes]
-version = "0.8"
-features = ["zeroize"]
-optional = true
-
-[dependencies.aes-gcm]
-version = "0.10"
-features = ["std"]
-optional = true
+#[dependencies.aes]
+#version = "0.8"
+#features = ["zeroize"]
+#optional = true
+#
+#[dependencies.aes-gcm]
+#version = "0.10"
+#features = ["std"]
+#optional = true
 
 [dependencies.anyhow]
 version = "1.0.18"
@@ -74,14 +74,14 @@ version = "1.0.18"
 [dependencies.base64]
 version = ">= 0.21, < 0.23"
 
-[dependencies.block-padding]
-version = "0.3"
-optional = true
-
-[dependencies.blowfish]
-version = "0.9"
-features = ["zeroize"]
-optional = true
+#[dependencies.block-padding]
+#version = "0.3"
+#optional = true
+#
+#[dependencies.blowfish]
+#version = "0.9"
+#features = ["zeroize"]
+#optional = true
 
 [dependencies.botan]
 version = "0.10.6"
@@ -95,85 +95,85 @@ default-features = false
 version = "0.4"
 optional = true
 
-[dependencies.camellia]
-version = "0.1"
-features = ["zeroize"]
-optional = true
-
-[dependencies.cast5]
-version = "0.11"
-features = ["zeroize"]
-optional = true
-
-[dependencies.cfb-mode]
-version = "0.8"
-optional = true
-
-[dependencies.cipher]
-version = "0.4"
-features = [
-    "std",
-    "zeroize",
-]
-optional = true
-
-[dependencies.des]
-version = "0.8"
-features = ["zeroize"]
-optional = true
-
-[dependencies.digest]
-version = "0.10"
-optional = true
-
-[dependencies.dsa]
-version = "0.6"
-optional = true
+#[dependencies.camellia]
+#version = "0.1"
+#features = ["zeroize"]
+#optional = true
+#
+#[dependencies.cast5]
+#version = "0.11"
+#features = ["zeroize"]
+#optional = true
+#
+#[dependencies.cfb-mode]
+#version = "0.8"
+#optional = true
+#
+#[dependencies.cipher]
+#version = "0.4"
+#features = [
+#    "std",
+#    "zeroize",
+#]
+#optional = true
+#
+#[dependencies.des]
+#version = "0.8"
+#features = ["zeroize"]
+#optional = true
+#
+#[dependencies.digest]
+#version = "0.10"
+#optional = true
+#
+#[dependencies.dsa]
+#version = "0.6"
+#optional = true
 
 [dependencies.dyn-clone]
 version = "1"
 
-[dependencies.eax]
-version = "0.5"
-optional = true
-
-[dependencies.ecb]
-version = "0.1"
-optional = true
-
-[dependencies.ecdsa]
-version = "0.16"
-features = [
-    "hazmat",
-    "arithmetic",
-]
-optional = true
-
-[dependencies.ed25519]
-version = "2"
-features = ["std"]
-optional = true
-default-features = false
-
-[dependencies.ed25519-dalek]
-version = "2"
-features = [
-    "rand_core",
-    "zeroize",
-]
-optional = true
+#[dependencies.eax]
+#version = "0.5"
+#optional = true
+#
+#[dependencies.ecb]
+#version = "0.1"
+#optional = true
+#
+#[dependencies.ecdsa]
+#version = "0.16"
+#features = [
+#    "hazmat",
+#    "arithmetic",
+#]
+#optional = true
+#
+#[dependencies.ed25519]
+#version = "2"
+#features = ["std"]
+#optional = true
+#default-features = false
+#
+#[dependencies.ed25519-dalek]
+#version = "2"
+#features = [
+#    "rand_core",
+#    "zeroize",
+#]
+#optional = true
 
 [dependencies.flate2]
 version = "1.0.1"
 optional = true
 
-[dependencies.idea]
-version = "0.5"
-features = ["zeroize"]
-optional = true
+#[dependencies.idea]
+#version = "0.5"
+#features = ["zeroize"]
+#optional = true
 
 [dependencies.idna]
-version = ">= 0.5, < 2"
+version = "0.4"
 
 [dependencies.lalrpop-util]
 version = "0.20"
@@ -184,10 +184,10 @@ version = "1.4.0"
 [dependencies.libc]
 version = "0.2.66"
 
-[dependencies.md-5]
-version = "0.10"
-features = ["oid"]
-optional = true
+#[dependencies.md-5]
+#version = "0.10"
+#features = ["oid"]
+#optional = true
 
 [dependencies.memsec]
 version = ">=0.5, <0.8"
@@ -213,38 +213,38 @@ optional = true
 version = "0.9.90"
 optional = true
 
-[dependencies.p256]
-version = "0.13"
-features = [
-    "ecdh",
-    "ecdsa",
-]
-optional = true
-
-[dependencies.p384]
-version = "0.13"
-features = [
-    "ecdh",
-    "ecdsa",
-]
-optional = true
-
-[dependencies.p521]
-version = "0.13"
-features = [
-    "ecdh",
-    "ecdsa",
-]
-optional = true
-
-[dependencies.rand]
-version = "0.8"
-optional = true
-default-features = false
-
-[dependencies.rand_core]
-version = "0.6"
-optional = true
+#[dependencies.p256]
+#version = "0.13"
+#features = [
+#    "ecdh",
+#    "ecdsa",
+#]
+#optional = true
+#
+#[dependencies.p384]
+#version = "0.13"
+#features = [
+#    "ecdh",
+#    "ecdsa",
+#]
+#optional = true
+#
+#[dependencies.p521]
+#version = "0.13"
+#features = [
+#    "ecdh",
+#    "ecdsa",
+#]
+#optional = true
+#
+#[dependencies.rand]
+#version = "0.8"
+#optional = true
+#default-features = false
+#
+#[dependencies.rand_core]
+#version = "0.6"
+#optional = true
 
 [dependencies.regex]
 version = "1"
@@ -252,45 +252,45 @@ version = "1"
 [dependencies.regex-syntax]
 version = "0.8"
 
-[dependencies.ripemd]
-version = "0.1"
-features = ["oid"]
-optional = true
-
-[dependencies.rsa]
-version = "0.9.0"
-optional = true
+#[dependencies.ripemd]
+#version = "0.1"
+#features = ["oid"]
+#optional = true
+#
+#[dependencies.rsa]
+#version = "0.9.0"
+#optional = true
 
 [dependencies.sha1collisiondetection]
 version = "0.3.1"
 features = ["std"]
 default-features = false
 
-[dependencies.sha2]
-version = "0.10"
-features = ["oid"]
-optional = true
+#[dependencies.sha2]
+#version = "0.10"
+#features = ["oid"]
+#optional = true
 
 [dependencies.thiserror]
 version = "1.0.2"
 
-[dependencies.twofish]
-version = "0.7"
-features = ["zeroize"]
-optional = true
-
-[dependencies.typenum]
-version = "1.12.0"
-optional = true
-
-[dependencies.x25519-dalek]
-version = "2"
-features = [
-    "static_secrets",
-    "zeroize",
-]
-optional = true
-default-features = false
+#[dependencies.twofish]
+#version = "0.7"
+#features = ["zeroize"]
+#optional = true
+#
+#[dependencies.typenum]
+#version = "1.12.0"
+#optional = true
+#
+#[dependencies.x25519-dalek]
+#version = "2"
+#features = [
+#    "static_secrets",
+#    "zeroize",
+#]
+#optional = true
+#default-features = false
 
 [dependencies.xxhash-rust]
 version = "0.8"
@@ -340,90 +340,90 @@ crypto-botan = [
     "botan?/botan3",
 ]
 crypto-botan2 = ["dep:botan"]
-crypto-cng = [
-    "dep:cipher",
-    "dep:eax",
-    "dep:winapi",
-    "dep:win-crypto-ng",
-    "dep:ed25519",
-    "dep:ed25519-dalek",
-    "dep:num-bigint-dig",
-    "dep:aes-gcm",
-    "dep:rand_core",
-]
+#crypto-cng = [
+#    "dep:cipher",
+#    "dep:eax",
+#    "dep:winapi",
+#    "dep:win-crypto-ng",
+#    "dep:ed25519",
+#    "dep:ed25519-dalek",
+#    "dep:num-bigint-dig",
+#    "dep:aes-gcm",
+#    "dep:rand_core",
+#]
 crypto-fuzzing = []
 crypto-nettle = ["dep:nettle"]
 crypto-openssl = [
     "dep:openssl",
     "dep:openssl-sys",
 ]
-crypto-rust = [
-    "dep:aes",
-    "dep:block-padding",
-    "dep:blowfish",
-    "dep:camellia",
-    "dep:cast5",
-    "dep:cfb-mode",
-    "dep:cipher",
-    "dep:des",
-    "dep:digest",
-    "dep:eax",
-    "dep:ecb",
-    "dep:ed25519",
-    "dep:ed25519-dalek",
-    "dep:idea",
-    "dep:md-5",
-    "dep:num-bigint-dig",
-    "dep:ripemd",
-    "dep:rsa",
-    "dep:sha2",
-    "sha1collisiondetection/digest-trait",
-    "sha1collisiondetection/oid",
-    "dep:twofish",
-    "dep:typenum",
-    "dep:x25519-dalek",
-    "dep:p256",
-    "dep:p384",
-    "dep:p521",
-    "dep:rand",
-    "rand?/getrandom",
-    "dep:rand_core",
-    "rand_core?/getrandom",
-    "dep:ecdsa",
-    "dep:aes-gcm",
-    "dep:dsa",
-]
+#crypto-rust = [
+#    "dep:aes",
+#    "dep:block-padding",
+#    "dep:blowfish",
+#    "dep:camellia",
+#    "dep:cast5",
+#    "dep:cfb-mode",
+#    "dep:cipher",
+#    "dep:des",
+#    "dep:digest",
+#    "dep:eax",
+#    "dep:ecb",
+#    "dep:ed25519",
+#    "dep:ed25519-dalek",
+#    "dep:idea",
+#    "dep:md-5",
+#    "dep:num-bigint-dig",
+#    "dep:ripemd",
+#    "dep:rsa",
+#    "dep:sha2",
+#    "sha1collisiondetection/digest-trait",
+#    "sha1collisiondetection/oid",
+#    "dep:twofish",
+#    "dep:typenum",
+#    "dep:x25519-dalek",
+#    "dep:p256",
+#    "dep:p384",
+#    "dep:p521",
+#    "dep:rand",
+#    "rand?/getrandom",
+#    "dep:rand_core",
+#    "rand_core?/getrandom",
+#    "dep:ecdsa",
+#    "dep:aes-gcm",
+#    "dep:dsa",
+#]
 default = [
     "compression",
     "crypto-nettle",
 ]
 
-[target."cfg(all(target_arch = \"wasm32\", target_os = \"unknown\"))".dependencies.chrono]
-version = "0.4.10"
-features = [
-    "std",
-    "wasmbind",
-    "clock",
-]
-default-features = false
-
-[target."cfg(all(target_arch = \"wasm32\", target_os = \"unknown\"))".dependencies.getrandom]
-version = "0.2"
-features = ["js"]
-
-[target."cfg(windows)".dependencies.win-crypto-ng]
-version = "0.5.1"
-features = [
-    "rand",
-    "block-cipher",
-]
-optional = true
-
-[target."cfg(windows)".dependencies.winapi]
-version = "0.3.8"
-features = ["bcrypt"]
-optional = true
-default-features = false
+#[target."cfg(all(target_arch = \"wasm32\", target_os = \"unknown\"))".dependencies.chrono]
+#version = "0.4.10"
+#features = [
+#    "std",
+#    "wasmbind",
+#    "clock",
+#]
+#default-features = false
+
+#[target."cfg(all(target_arch = \"wasm32\", target_os = \"unknown\"))".dependencies.getrandom]
+#version = "0.2"
+#features = ["js"]
+
+#[target."cfg(windows)".dependencies.win-crypto-ng]
+#version = "0.5.1"
+#features = [
+#    "rand",
+#    "block-cipher",
+#]
+#optional = true
+
+#[target."cfg(windows)".dependencies.winapi]
+#version = "0.3.8"
+#features = ["bcrypt"]
+#optional = true
+#default-features = false
 
 [badges.gitlab]
 repository = "sequoia-pgp/sequoia"
