--- a/src/clipboard.rs
+++ b/src/clipboard.rs
@@ -142,7 +142,7 @@
     if let Some(RawDisplayHandle::Wayland(display)) = raw_display_handle {
         log::trace!("Initializing smithay clipboard…");
         #[allow(unsafe_code)]
-        Some(unsafe { smithay_clipboard::Clipboard::new(display.display.as_ptr()) })
+        Some(unsafe { smithay_clipboard::Clipboard::new(display.display) })
     } else {
         #[cfg(feature = "wayland")]
         log::debug!("Cannot init smithay clipboard without a Wayland display handle");
