--- a/src/text/text_layout.rs
+++ b/src/text/text_layout.rs
@@ -1032,6 +1032,7 @@
         }
     }
 
+    #[ignore = "broken in Debian"]
     #[test]
     fn test_cjk() {
         let mut fonts = FontsImpl::new(1.0, 1024, FontDefinitions::default());
@@ -1047,6 +1048,7 @@
         );
     }
 
+    #[ignore = "broken in Debian"]
     #[test]
     fn test_pre_cjk() {
         let mut fonts = FontsImpl::new(1.0, 1024, FontDefinitions::default());
